contracts/pcv/morpho/MorphoCompoundPCVDeposit.sol

```diff
@@ -87,33 +119,120 @@ contract MorphoCompoundPCVDeposit is PCVDeposit {
     /// @notice withdraw tokens from the PCV allocation
     /// @param to the address PCV will be sent to
     /// @param amount of tokens withdrawn
-    function withdraw(address to, uint256 amount) external onlyPCVController {
+    function withdraw(address to, uint256 amount)
+        external
+        onlyPCVController
+        nonReentrant
+    {
         _withdraw(to, amount);
     }
 
     /// @notice withdraw all tokens from Morpho
     /// @param to the address PCV will be sent to
-    function withdrawAll(address to) external onlyPCVController {
-        uint256 amount = balance();
-        _withdraw(to, amount);
+    function withdrawAll(address to) external onlyPCVController nonReentrant {
+        /// compute profit from interest accrued and emit an event
+        _recordPNL();
+
+        /// withdraw deposited amount as this was updated in record pnl
+        _withdraw(to, depositedAmount);
```

Comment:
the `_withdraw` internal function already calls `_recordPNL`, I understand why you want to update the indexes and the `depositedAmount` state variable before calling `withdraw` with `depositedAmount` parameter, but then maybe add a boolean 3rd param to the internal function to skip the _reportPNL() call in it